src/components/backgroundselector/BackgroundSelectorDirective.js

```diff
@@ -91,14 +94,26 @@ goog.require('ga_topic_service');
           };
 
           var dereg = scope.$on('gaLayersChange', function(event, newLayers) {
+            getLayersDeffered.resolve();
             updateBgLayer();
             dereg();
```

Comment:
This code blocks come from your (not you personally) current implementation. I think that the goal is to override the default bg layer from the topic with the one the user selected (which only happens once, hence the dereg). However, with my current implementation, it is always the default bg layer which is chosen since I can only be sure that the layers are loaded once `gaLayersChange` is emitted.

I will wait on your PR addressing init to fix this correctly and clear the code of this PR.